Features/output

Stalfy opened the pull request:
TODO : Replace declared stubs with real classes stubs once they are implemented.

Stalfy committed:
Added basic md conversion.

Stalfy committed:
Completed basic test conversion to html.

Stalfy committed:
Modified tests to stay in scope.

Stalfy committed:
Completed successful test formatting.

Stalfy committed:
Added failure formatting.

Stalfy committed:
Completed basic output generation.

Stalfy committed:
Commented out of scope elements.

Grasseh commented on src/output/generator.js:
```diff
@@ -0,0 +1,103 @@
+
+        let relativeCssPath = path.relative(path.dirname(mdFilePath), cssFilePath);
+
+        let htmlContent = '<!DOCTYPE html>\n<html>\n<head>\n';
```
Would using a templating engine such as handlebars be better for this and the next four lines?

doomy23 replied:
We could do that indeed in the next iteration maybe tho?

Stalfy replied:
Let's talk it out in the next meeting.

Stalfy replied:
TBH
...cssPaths + forEach could do the trick.

Grasseh reviewed (approved):
LGTM although you may prefer using some form of templating engine just to facilitate the HTML frames. Mostly because these frames may grow in the future (more styles/scripts in the head), and we don't want a JS file with 50 htmlContent += stuff

Stalfy committed:
Reworked output to protect from major flaw.
If multiple tests had the same expected result and test name, all would be replaced.

Lolskate commented on test/unit/artifacts/output/FailingTests.md:
```diff
@@ -0,0 +1,2 @@
+This is a [failing](?=failingTest()) test.  
```
Dans le markdown que j'utilise pour le parser je mets un "&" devant un résultat de test et j'ai le nom de la classe avant la méthode: "[&failing](?=myClass.failingTest())". Je pense pas que le "&" change quelque chose, par contre maintenant le "testName" devient "Class.function()". On vérifiera ça pour la prochaine itération.

Lolskate commented on src/output/generator.js:
```diff
@@ -0,0 +1,98 @@
+
+class HTMLGenerator {
+    constructor() {
+        this.testFormatting = Object.freeze({
```
Je connais pas Object.freeze, mais en gros c'est un pointeur de fonction que tu fais avec? - pas de commentaire négatif, juste curieux d'apprendre la dessus.
